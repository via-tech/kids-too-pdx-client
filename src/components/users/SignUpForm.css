.SignUp {
  label, input {
    display: block;
    margin-top: 1em;
  }

  input[type="radio"] {
    display: inline;
  }

  .error {
    color: red;
  }
}