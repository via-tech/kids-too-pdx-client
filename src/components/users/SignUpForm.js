import React from 'react';
import PropTypes from 'prop-types';

function SignUpForm({ handleSubmit, handleChange }) {
  return (
    <form onSubmit={handleSubmit}>
      <fieldset>
        <legend>Organization Info</legend>
        <label>Username:
          <input type="text" name="username" onChange={handleChange} required />
        </label>
        <label>Organization Name:
          <input type="text" name="name" onChange={handleChange} required/>
        </label>
        <label>Email:
          <input type="email" name="email" onChange={handleChange} required />
        </label>
        <label>Phone:
          <input type="tel" name="phone" onChange={handleChange} required />
        </label>
        <label>Address:
          <label>Street:
            <input type="text" name="street" onChange={handleChange} />
          </label>
          <label>City:
            <input type="text" name="city" onChange={handleChange} />
          </label>
          <label>Zip Code:
            <input type="text" name="zipcode" pattern="[0-9]*" onChange={handleChange} />
          </label>
        </label>
        <label>Organization Bio:
          <textarea name="bio" onChange={handleChange} />
        </label>
        <label>Password:
          <input type="text" name="password" onChange={handleChange} required />
        </label>
      </fieldset>
      <button type="submit">Sign Up</button>
    </form>
  );
}

SignUpForm.propTypes = {
  handleSubmit: PropTypes.func,
  handleChange: PropTypes.func
};

export default SignUpForm;
