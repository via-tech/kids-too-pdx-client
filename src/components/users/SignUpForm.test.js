import React from 'react';
import renderer from 'react-test-renderer';
import SignUpForm from './SignUpForm';

describe('Sign up form', () => {
  it('snapshots SignUpForm', () => {
    const props = {
      handleSubmit: jest.fn(),
      handleChange: jest.fn()
    };

    const tree = renderer.create(
      <SignUpForm {...props} />
    );

    expect(tree).toMatchSnapshot();
  });
});
