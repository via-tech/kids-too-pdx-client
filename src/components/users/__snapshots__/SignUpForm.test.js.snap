// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign up form snapshots SignUpForm 1`] = `
<form
  className="SignUp"
  onSubmit={[Function]}
>
  <fieldset>
    <legend>
      Organization Info
    </legend>
    <label>
      Username
      <span
        className="required"
      >
        *
      </span>
      <input
        name="username"
        onChange={[MockFunction]}
        required={true}
        type="text"
      />
    </label>
    <label>
      Organization Name
      <span
        className="required"
      >
        *
      </span>
      <input
        name="name"
        onChange={[MockFunction]}
        required={true}
        type="text"
      />
    </label>
    <label>
      Email
      <span
        className="required"
      >
        *
      </span>
      <input
        name="email"
        onChange={[MockFunction]}
        required={true}
        type="email"
      />
    </label>
    <label>
      Phone
      <span
        className="required"
      >
        *
      </span>
      <input
        name="phone"
        onChange={[MockFunction]}
        required={true}
        type="tel"
      />
    </label>
    <label>
      Address
      <span
        className="required"
      >
        *
      </span>
      <label>
        Street
        <span
          className="required"
        >
          *
        </span>
        <input
          name="street"
          onChange={[MockFunction]}
          required={true}
          type="text"
        />
      </label>
      <label>
        City
        <span
          className="required"
        >
          *
        </span>
        <input
          name="city"
          onChange={[MockFunction]}
          required={true}
          type="text"
        />
      </label>
      <label>
        State
        <span
          className="required"
        >
          *
        </span>
        <input
          name="state"
          onChange={[MockFunction]}
          required={true}
          type="text"
        />
      </label>
      <label>
        Zip Code
        <span
          className="required"
        >
          *
        </span>
        <input
          name="zipcode"
          onChange={[MockFunction]}
          pattern="[0-9]*"
          required={true}
          type="text"
        />
      </label>
    </label>
    <label>
      Organization Bio:
      <textarea
        name="bio"
        onChange={[MockFunction]}
      />
    </label>
    <label>
      Password
      <span
        className="required"
      >
        *
      </span>
      <input
        name="password"
        onChange={[MockFunction]}
        required={true}
        type="password"
      />
    </label>
    <label>
      Confirm Password
      <span
        className="required"
      >
        *
      </span>
      <input
        name="confirmPassword"
        onChange={[MockFunction]}
        required={true}
        type="password"
      />
    </label>
  </fieldset>
  <fieldset>
    <legend>
      Payment Info
    </legend>
    <label>
      Card Number
      <span
        className="required"
      >
        *
      </span>
      <input
        maxLength="19"
        name="cardNumber"
        onChange={[MockFunction]}
        pattern="[0-9]*"
        placeholder="Card Number"
        required={true}
        type="text"
      />
    </label>
    <label>
      Security Code
      <span
        className="required"
      >
        *
      </span>
      <input
        maxLength="3"
        name="securityCode"
        onChange={[MockFunction]}
        required={true}
        type="text"
      />
    </label>
    <label>
      Name
      <span
        className="required"
      >
        *
      </span>
      <input
        name="cardName"
        onChange={[MockFunction]}
        placeholder="Card Name"
        required={true}
        type="text"
      />
    </label>
    <label>
      Expiration Date
      <span
        className="required"
      >
        *
      </span>
      <input
        maxLength="2"
        name="expMonth"
        onChange={[MockFunction]}
        pattern="[0-9]*"
        placeholder="mm"
        required={true}
        type="text"
      />
      <input
        maxLength="4"
        minLength="4"
        name="expYear"
        onChange={[MockFunction]}
        pattern="[0-9]*"
        placeholder="yyyy"
        required={true}
        type="text"
      />
    </label>
    <label>
      Method
      <span
        className="required"
      >
        *
      </span>
      <input
        name="method"
        onChange={[MockFunction]}
        required={true}
        type="radio"
        value="visa"
      />
      Visa
      <input
        name="method"
        onChange={[MockFunction]}
        required={true}
        type="radio"
        value="master"
      />
      Master
    </label>
    <label>
      Billing Address 
      <span
        className="required"
      >
        *
      </span>
      <label>
        Street:
        <input
          name="billStreet"
          onChange={[MockFunction]}
          type="text"
        />
      </label>
      <label>
        City:
        <input
          name="billCity"
          onChange={[MockFunction]}
          type="text"
        />
      </label>
      <label>
        State
        <input
          name="state"
          onChange={[MockFunction]}
          required={true}
          type="text"
        />
      </label>
      <label>
        Zip Code:
        <input
          name="billZipcode"
          onChange={[MockFunction]}
          pattern="[0-9]*"
          type="text"
        />
      </label>
    </label>
  </fieldset>
  <h2>
    Total: $
  </h2>
  <button
    type="submit"
  >
    Sign Up
  </button>
</form>
`;

exports[`Sign up form snapshots form with error 1`] = `
<form
  className="SignUp"
  onSubmit={[Function]}
>
  <p
    className="error"
  >
    User already exists
  </p>
  <fieldset>
    <legend>
      Organization Info
    </legend>
    <label>
      Username
      <span
        className="required"
      >
        *
      </span>
      <input
        name="username"
        onChange={[MockFunction]}
        required={true}
        type="text"
      />
    </label>
    <label>
      Organization Name
      <span
        className="required"
      >
        *
      </span>
      <input
        name="name"
        onChange={[MockFunction]}
        required={true}
        type="text"
      />
    </label>
    <label>
      Email
      <span
        className="required"
      >
        *
      </span>
      <input
        name="email"
        onChange={[MockFunction]}
        required={true}
        type="email"
      />
    </label>
    <label>
      Phone
      <span
        className="required"
      >
        *
      </span>
      <input
        name="phone"
        onChange={[MockFunction]}
        required={true}
        type="tel"
      />
    </label>
    <label>
      Address
      <span
        className="required"
      >
        *
      </span>
      <label>
        Street
        <span
          className="required"
        >
          *
        </span>
        <input
          name="street"
          onChange={[MockFunction]}
          required={true}
          type="text"
        />
      </label>
      <label>
        City
        <span
          className="required"
        >
          *
        </span>
        <input
          name="city"
          onChange={[MockFunction]}
          required={true}
          type="text"
        />
      </label>
      <label>
        State
        <span
          className="required"
        >
          *
        </span>
        <input
          name="state"
          onChange={[MockFunction]}
          required={true}
          type="text"
        />
      </label>
      <label>
        Zip Code
        <span
          className="required"
        >
          *
        </span>
        <input
          name="zipcode"
          onChange={[MockFunction]}
          pattern="[0-9]*"
          required={true}
          type="text"
        />
      </label>
    </label>
    <label>
      Organization Bio:
      <textarea
        name="bio"
        onChange={[MockFunction]}
      />
    </label>
    <label>
      Password
      <span
        className="required"
      >
        *
      </span>
      <input
        name="password"
        onChange={[MockFunction]}
        required={true}
        type="password"
      />
    </label>
    <label>
      Confirm Password
      <span
        className="required"
      >
        *
      </span>
      <input
        name="confirmPassword"
        onChange={[MockFunction]}
        required={true}
        type="password"
      />
    </label>
  </fieldset>
  <fieldset>
    <legend>
      Payment Info
    </legend>
    <label>
      Card Number
      <span
        className="required"
      >
        *
      </span>
      <input
        maxLength="19"
        name="cardNumber"
        onChange={[MockFunction]}
        pattern="[0-9]*"
        placeholder="Card Number"
        required={true}
        type="text"
      />
    </label>
    <label>
      Security Code
      <span
        className="required"
      >
        *
      </span>
      <input
        maxLength="3"
        name="securityCode"
        onChange={[MockFunction]}
        required={true}
        type="text"
      />
    </label>
    <label>
      Name
      <span
        className="required"
      >
        *
      </span>
      <input
        name="cardName"
        onChange={[MockFunction]}
        placeholder="Card Name"
        required={true}
        type="text"
      />
    </label>
    <label>
      Expiration Date
      <span
        className="required"
      >
        *
      </span>
      <input
        maxLength="2"
        name="expMonth"
        onChange={[MockFunction]}
        pattern="[0-9]*"
        placeholder="mm"
        required={true}
        type="text"
      />
      <input
        maxLength="4"
        minLength="4"
        name="expYear"
        onChange={[MockFunction]}
        pattern="[0-9]*"
        placeholder="yyyy"
        required={true}
        type="text"
      />
    </label>
    <label>
      Method
      <span
        className="required"
      >
        *
      </span>
      <input
        name="method"
        onChange={[MockFunction]}
        required={true}
        type="radio"
        value="visa"
      />
      Visa
      <input
        name="method"
        onChange={[MockFunction]}
        required={true}
        type="radio"
        value="master"
      />
      Master
    </label>
    <label>
      Billing Address 
      <span
        className="required"
      >
        *
      </span>
      <label>
        Street:
        <input
          name="billStreet"
          onChange={[MockFunction]}
          type="text"
        />
      </label>
      <label>
        City:
        <input
          name="billCity"
          onChange={[MockFunction]}
          type="text"
        />
      </label>
      <label>
        State
        <input
          name="state"
          onChange={[MockFunction]}
          required={true}
          type="text"
        />
      </label>
      <label>
        Zip Code:
        <input
          name="billZipcode"
          onChange={[MockFunction]}
          pattern="[0-9]*"
          type="text"
        />
      </label>
    </label>
  </fieldset>
  <h2>
    Total: $
  </h2>
  <button
    type="submit"
  >
    Sign Up
  </button>
</form>
`;
